client/packages/system/src/Stock/Components/StatusHistory/StatusHistory.tsx

```diff
@@ -20,19 +25,57 @@ export const StatusHistory = ({
 }: StatusHistoryProps): ReactElement => {
   const t = useTranslation();
   const { columns } = useStatusHistoryColumns();
+  const [selectedStatusLog, setSelectedStatusLog] =
+    useState<VvmStatusLogRowFragment>();
+  const { onOpen, onClose, isOpen } = useEditModal<DraftStockLine>();
 
-  const vvmStatusLogs = draft?.vvmStatusLogs?.nodes ?? [];
+  const handleAddClick = () => {
+    setSelectedStatusLog(undefined);
+    onOpen();
+  };
+
+  const handleRowClick = (row: VvmStatusLogRowFragment) => {
+    setSelectedStatusLog(row);
+    onOpen();
+  };
 
   return (
     <TableProvider createStore={createTableStore}>
-      <DataTable
-        id="stockline-status-history"
-        columns={columns}
-        data={vvmStatusLogs}
-        isLoading={isLoading}
-        noDataMessage={t('messages.no-status-history')}
-        overflowX="auto"
-      />
+      <Box width="100%">
+        <DataTable
+          id="stockline-status-history"
+          columns={columns}
+          data={draft?.vvmStatusLogs?.nodes ?? []}
+          onRowClick={handleRowClick}
+          isLoading={isLoading}
+          noDataMessage={t('messages.no-status-history')}
+          overflowX="auto"
+        />
+        <Box
+          sx={{
+            p: 2,
+            zIndex: 2,
+            position: 'sticky',
```

Comment:
Nice this keeps the button visible! Wasn't going to say it didn't matter too much but this is great.